### PR TITLE
[PF-2186] replace mock `ControlledResourceMetadataManager` and `ControlledResourceService` with autowired in unit test

### PR DESCRIPTION
**Pull request by ginaygoogle**
This PR comes from the ticket [PF-2186](https://broadworkbench.atlassian.net/browse/PF-2186). 

The aim for this ticket is remove the mock logic for `ControlledResourceMetadataManager` and `ControlledResourceService` in favor of the real ones. and then they can be tested.

### CODE REVIEW

File: service/src/test/java/bio/terra/workspace/app/controller/CreateAzureStorageContainerSasTokenTest.java [hunk 1]
```diff
@@ -37,13 +44,21 @@ public class CreateAzureStorageContainerSasTokenTest extends BaseAzureUnitTest {
...
+  @Autowired WorkspaceDao workspaceDao;
+  @Autowired WorkspaceService workspaceService;
+  @Autowired ResourceDao resourceDao;
+  @Autowired SpendConnectedTestUtils spendUtils;
 
   @BeforeEach
   public void setup() throws Exception {
     workspaceId = UUID.randomUUID();
+    var azureCloudContext = new AzureCloudContext("fake", "fake", "fake");
+    WorkspaceUnitTestUtils.createWorkspaceWithAzureContext(
```
[the remark this answers is not in the document]
Comment (yuhuyoyo) [reply service/src/test/java/bio/terra/workspace/app/controller/CreateAzureStorageContainerSasTokenTest.java:57]: I also like the former. it's what createWorkspaceWithGcpContext does too.

File: service/src/test/java/bio/terra/workspace/service/resource/controlled/flight/clone/bucket/CopyGcsBucketDefinitionStepTest.java [hunk 2]
```diff
@@ -121,22 +172,11 @@ public void testDoStep() throws InterruptedException {
     assertEquals(SOURCE_BUCKET_RESOURCE.getManagedBy(), destinationBucketResource.getManagedBy());
     assertEquals(
         SOURCE_BUCKET_RESOURCE.getApplicationId(), destinationBucketResource.getApplicationId());
-    assertEquals(
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/workspace/service/resource/controlled/flight/clone/bucket/CopyGcsBucketDefinitionStepTest.java:124]: why is this deleted?

File: service/src/test/java/bio/terra/workspace/service/resource/controlled/flight/clone/bucket/CopyGcsBucketDefinitionStepTest.java [hunk 3]
```diff
@@ -47,18 +62,67 @@
...
   @Mock FlightContext mockFlightContext;
+  @Mock private StorageCow mockStorageCow;
+  @Mock private BucketCow mockBucketCow;
+  @Mock private Storage mockStorageClient;
+  @Mock private Storage.Buckets mockBuckets;
+  @Mock private Storage.Buckets.Get mockStorageBucketsGet;
+  @Mock private GcpCloudContext mockGcpCloudContext;
+  @Mock private Policy mockPolicy;
+
+  @Captor private ArgumentCaptor<BucketInfo> bucketInfoCaptor;
```
Comment (yuhuyoyo) [inline service/src/test/java/bio/terra/workspace/service/resource/controlled/flight/clone/bucket/CopyGcsBucketDefinitionStepTest.java:77]: were you planning to use this? i didn't see you assert the captured value below.

File: service/src/test/java/bio/terra/workspace/service/resource/controlled/flight/clone/bucket/CopyGcsBucketDefinitionStepTest.java (hunk 2 above)
Comment (yuhuyoyo) [reply service/src/test/java/bio/terra/workspace/service/resource/controlled/flight/clone/bucket/CopyGcsBucketDefinitionStepTest.java:124]: ah thanks for explaining. this's an example of why mocking is bad. :)  yeah let's assert the current pricate state is ACTIVE.

**Review by ddietterich — APPROVED** [review]
LGTM
